ui/app/pages/routes/index.js

```diff
@@ -98,6 +98,27 @@ class Routes extends Component {
     })
   }
 
+  componentDidUpdate (prevProps) {
+    const {
+      threeBoxSynced,
+      show3BoxRestoreConfirmModal,
+      getThreeBoxLastUpdated,
+      setThreeBoxSyncingPermission,
+      restoredFromThreeBox,
+    } = this.props
+
+    if (!prevProps.threeBoxSynced && threeBoxSynced && restoredFromThreeBox === null) {
```

Comment:
I'm not clear on what purpose `restoredFromThreeBox` serves. It looks like the primary purpose is to prevent showing the restore confirm modal more than once in a session. However, the only case where `threeBoxSynced` could conceivably be switching from false to true is for the initial sync, unless 3box was toggled off then on again.

Is there any harm in showing the modal again after 3Box is re-enabled though? It might be useful to give the user some feedback about when syncing has finished. I'm not sure.

Reply:
The reason that we don't want to show the modal after 3Box is re-enabled is that we want to prevent actual restoring/overwriting of local data with remote data at all times other then after import. This is to prevent a case such as: local state is A -> threebox is turned off -> local state changes to B -> threebox is turned on -> modal is shown and confirm -> local state is set back to A

We could, however, show a different modal when it is re-enabled / toggled on. Probably not necessary at this time.

Reply:
Moreover, `restoredFromThreeBox` persists across sessions, whereas `threeBoxSynced` defaults to false for every new session. So `restoredFromThreeBox` tells us whether the user did in fact approval the overwriting of local data with remote data after an import. This ends up preventing the modal from showing up more than "once per import".

Reply:
This all makes sense; I somehow got confused about what happened upon confirmation.